agent/consul/rate/handler.go

```diff
@@ -130,8 +137,13 @@ type HandlerConfig struct {
 
 	// GlobalReadConfig configures the global rate limiter for read operations.
 	GlobalReadConfig multilimiter.LimiterConfig
+
+	// makeLimiter is a function that can be passed by tests to inject a mock
+	// multilimiter (it is unexported so it cannot be used in production code).
+	makeLimiter func(multilimiter.Config) multilimiter.RateLimiter
 }
 
+//go:generate mockery --name HandlerDelegate --inpackage --filename mock_HandlerDelegate_test.go
 type HandlerDelegate interface {
```

Comment:
Personal preference so feel free to ignore but the name is a bit confusing

Reply:
Agreed, suggestions for better names welcome 😄 .

Reply:
Yeah I'm really bad at naming things 😆 

What about `ServerRoleDelegate` or `LeadershipCheckDelegate`?

Reply:
Renamed HandlerDelegate -> LeaderStatusProvider in https://github.com/hashicorp/consul/commit/02ca05b195697814bce41cfad10ace4322711eb8